src/Pim/Bundle/DataGridBundle/Normalizer/ProductAssociationNormalizer.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Pim\Bundle\DataGridBundle\Normalizer;
+
+use Akeneo\Component\StorageUtils\Exception\InvalidObjectException;
+use Pim\Component\Catalog\Model\ProductInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\SerializerAwareNormalizer;
+
+/**
+ * Product association normalizer for datagrid
+ *
+ * @author    Marie Bochu <marie.bochu@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductAssociationNormalizer extends SerializerAwareNormalizer implements NormalizerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($product, $format = null, array $context = [])
+    {
+        if (!$this->serializer instanceof NormalizerInterface) {
```

Comment:
why in this specific case you need to check the serialiser and not for every other normalizers?